packages/crypto/src/SparseMerkleTree.ts

```diff
@@ -22,44 +20,38 @@ const newWrappedPoseidonT3Hash = (
 }
 
 export class SparseMerkleTree {
-    protected root?: BigInt
-    private zeroHashes!: BigInt[]
-
-    public readonly numLeaves: BigInt
-
-    public static async create(
-        db,
-        height: number,
-        zeroHash: BigInt
-    ): Promise<SparseMerkleTree> {
-        const tree = new SparseMerkleTree(db, height)
-        await tree.init(zeroHash)
-        return tree
-    }
+    protected _root: bigint
+    private zeroHashes!: bigint[]
+
+    public readonly numLeaves: bigint
 
-    constructor(protected db, private height: number) {
-        assert(height > 0, 'SMT height needs to be > 0')
+    constructor(protected db, private _height: number, zeroHash: bigint) {
```

Comment:
I'm not sure what `db` argument is here. I know that a `keyv` instance is passed here but not much else. I don't like `keyv` because it doesn't compile well for browsers, it has a lot of nodejs specific requirements.

Could we instead use the `zk-kit` `SparseMerkleTree` implementation?

Reply:
Their implementation is really different. I discussed with them but haven't found a solution to use the package. So I just kept the old one.

yeah we can remove `keyv`